### DISCUSSION TITLE
Allow relative to buffer config in pasteImage

### DISCUSSION THREAD

**Discussion by codingluke** [epwalsh/obsidian.nvim#365] [Ideas]
Hi there. I very much love this plugin. Even though I am not to much into Obsidian, the daily note feature I use almost every day :)
And also the `:ObsidianPastImg` command is a very handy alternative for https://github.com/ekickx/clipboard-image.nvim, which seems abondend and is not working on the latest neovim anymore.

As I am also maintaining a lot of docusaurus pages, it would be great to be able to declare a path **relative to the current buffer.** This would make the plugin even more flexible for over all markdown editing.

What do you think?

**Comment by epwalsh** [comment]
Hey @codingluke, this will be possible (without hackery) once https://github.com/epwalsh/obsidian.nvim/pull/366 merges by adding a  "dynamic" workspace to your config.

**Reply by epwalsh** [reply]
Alternatively you can call obsidian.nvim's image pasting function directly:

```lua
local paste_img = require("obsidian.img_paste").paste_img

-- Save the image to disk and get the absolute path.
local abs_image_path = assert(paste_img(filename, directory))

-- Insert a markdown image link with a path relative to the current buffer.
local rel_image_path = abs_image_path:make_relative(vim.api.nvim_buf_get_name(0))
vim.api.nvim_put({ "![alt text](" .. rel_image_path .. ")" }, "", false, true)
```

**Reply by codingluke** [reply]
Hi @epwalsh thank you very much!
